src/Data/Lens/Prism.purs

```diff
@@ -18,23 +57,78 @@ import Data.Profunctor (dimap, rmap)
 import Data.Profunctor.Choice (right)
 import Data.Newtype (under)
 
--- | Create a `Prism` from a constructor/pattern pair.
+-- | Create a `Prism` from a constructor and a "focus" function that
```

Comment:
Might be worth commenting on the `Either t` bit here? Specifically, it wasn't obvious to me why there was a `t` there at all.

Reply:
Also, it might be worth talking about the prism laws briefly here?

Reply:
I don't understand why it's `Either t a` instead of `Either somethingElse a`, so I can't explain it. 

Regarding laws: I do cover laws in [*Lenses for the Mere Mortal*](https://leanpub.com/lenses), in more detail than can be done in an API document. Provided I continue to make it free, that seems a better place to talk about laws.

Reply:
I think you could explain it succinctly as

```
_Note_: The matching function returns a result wrapped in `Either t` so as to allow for type-changing prisms in the case where the input does not match.
```

What do you think?

Reply:
I disagree about the laws actually. I think they ought to be documented with the types themselves, like we do for type classes in Prelude.

Reply:
This isn't to say we need to add them here _now_, just that I'd like to eventually, and that redirecting readers to an external resource from doc comments doesn't strike me as ideal.

Reply:
I do refer to the laws section of the Haskell `lens` fairly often, so I take your point.

I tend to explain laws in words, then equational and example style:

```
set-get:

: `view` retrieves what `set` puts in. 
  
  {lang=haskell}
  ~~~~~~~~~~~~~~
  set lens >>> view lens ≡ id
  
  > set first "NEW" (Tuple 1 2) # view first
  "NEW"
  ~~~~~~~~~~~~~~
```

That OK?

Reply:
Yeah I like that.

Reply:
How do I explain the `Traversal` laws? The [Haskell traversal laws](https://hackage.haskell.org/package/lens-4.16.1/docs/Control-Lens-Traversal.html#g:1) don't translate into profunctor lenses, and it seems odd to write:

```
traverseOf traversed pure ≡ pure
```

Reply:
```purescript
over traversed pure ≡ map pure
```
?